runtime/src/bank_client.rs

```diff
@@ -197,6 +197,7 @@ impl BankClient {
                 transactions.push(tx);
             }
             let _ = bank.process_transactions(&transactions);
+            bank.commit_credits();
```

Comment:
I noticed this bank above is an Arc, there aren't any parallel transactions being processed on this bank are there? Worried that the clearing of the credit only account in this call to `commit_credits` might corrupt any parallel processing on this bank